testsuite/tests/ui/mail_and_messages/test_automatic_mails.py

```diff
@@ -0,0 +1,47 @@
+"""Test of automatic mails functionality in UI"""
+import re
+import pytest
+
+
+from testsuite.ui.views.admin.audience.account import AccountInvitationNewView
+from testsuite.ui.views.devel.login import InvitationSignupView
+from testsuite.utils import blame
+
+
+# pylint: disable=too-many-arguments
+@pytest.mark.usefixtures("login")
+def test_invitation_mail(account, navigator, mailhog_client, provider_account, request, custom_devel_login, browser):
+    """
+    Test:
+        - Send invitation to new user
+        - Assert that invitation were sent via mail
+        - Signup with email signup link
+        - Login with new signed-up user
+        - Assert that login was successful
+    """
+    email = f"{blame(request, 'mail')}@anything.invalid"
+    username = blame(request, "user")
+    password = blame(request, "passwd")
+
+    invitation = navigator.navigate(AccountInvitationNewView, account=account)
+    invitation.invite_user(email)
+
+    # assert inside mailhog method
+    mailhog_client.assert_message_received(
+        subject=f"Invitation to join {account.entity_name}", receiver=email, expected_count=1
+    )
+    mail = mailhog_client.find_message(subject=f"Invitation to join {account.entity_name}", receiver=email)
```

Comment:
Actually this could be returned by `assert_message_received`. It can be bit unusual for assert to return something, anyway this means same call twice -> twice as much time to run the test.

Reply:
Or alternatively, you could `find_message` first and assert could work with already fetched mails like this:
```
mail = find_message(...)
assert_message_received(mail, expected_count=1)
```

Reply:
@mganisin It doesn't mean twice much time as fetching mails is only a little part of the test time, but I am not sure about the correct solution as well